src/Npgsql.GeoJSON/GeoJSONHandler.cs

```diff
@@ -124,48 +125,48 @@ static Exception AllOrNoneCoordiantesMustHaveZ(NpgsqlParameter? parameter, strin
 
         #region Read
 
-        public override ValueTask<GeoJSONObject> Read(NpgsqlReadBuffer buf, int len, bool async, FieldDescription? fieldDescription = null)
-            => ReadGeometry(buf, async);
+        public override ValueTask<GeoJSONObject> Read(NpgsqlReadBuffer buf, int len, bool async, FieldDescription? fieldDescription = null, CancellationToken cancellationToken = default)
```

Comment:
@roji as you can see, it looks bad. Handlers **must** accept a cancellation token since it's always there.

Reply:
Also, there are places where the compiler gives a warning CS1066, if the interface already has a default value.
For example:
```csharp
async ValueTask<Point> INpgsqlTypeHandler<Point>.Read(NpgsqlReadBuffer buf, int len, bool async, FieldDescription? fieldDescription, CancellationToken cancellationToken)
```

Reply:
I don't get it, maybe I'm missing something... what looks bad exactly? This looks like the standard C# way of writing a method...

Reply:
That methods **_always_** takes a cancellation token.

Reply:
@vonzshik right - implementations of an interface method are indeed supposed to omit the default. That's a case of its own.

@YohDeadfall I'm sorry, I disagree. Even if this specific method happens to always get passed a cancellation token, in our *current* codebase, other methods don't (because they're called by sync-only methods). If those other methods have an optional cancellation token, then we're just inconsistent for no reason (they will have the cancellation token last, this one won't).

I'll turn it around in a different way. If we insist on making the token non-optional, then it can no longer come last, which is another C# convention (and code analysis warning) that we would have to break. So in order for our code to follow coding standards, the parameter needs to be optional.

To summarize my opinion, given that the compiler enforces flowing the token, we gain nothing from making it non-optional - no safety or code quality is increased. All we do is make our code non-standard.

Reply:
Re the runtime/.NET not respecting this, here's the issue for .NET 5.0: https://github.com/dotnet/runtime/issues/33774. And here's the analyzer issue for it: https://github.com/dotnet/roslyn-analyzers/pull/3641. I volunteer to make sure the analyzer works properly and identifies any problematic cases.

Reply:
Actually it seems like we get these warnings for free via the SDK - probably starting in 5.0.0-rc2 - since we've added `<AnalysisLevel>latest</AnalysisLevel>` in #3122.

See https://github.com/dotnet/roslyn-analyzers#microsoftcodeanalysisnetanalyzers for more details.